src/include/homestore/btree/btree.hpp

```diff
@@ -32,16 +32,42 @@ SISL_LOGGING_DECL(btree)
 namespace homestore {
 
 using BtreeNodePtr = boost::intrusive_ptr< BtreeNode >;
+using BtreeNodeList = folly::small_vector< BtreeNodePtr, 3 >;
 
 struct BtreeThreadVariables {
     std::vector< btree_locked_node_info > wr_locked_nodes;
     std::vector< btree_locked_node_info > rd_locked_nodes;
     BtreeNodePtr force_split_node{nullptr};
 };
 
+struct BTREE_FLIPS {
+    static constexpr uint32_t INDEX_PARENT_NON_ROOT = 1 << 0;
+    static constexpr uint32_t INDEX_PARENT_ROOT = 1 << 1;
+    static constexpr uint32_t INDEX_LEFT_SIBLING = 1 << 2;
+    static constexpr uint32_t INDEX_RIGHT_SIBLING = 1 << 3;
+
+    uint32_t flips;
+    BTREE_FLIPS() : flips{0} {}
+    std::string list() const {
+        std::string str;
+        if (flips & INDEX_PARENT_NON_ROOT) { str += "index_parent_non_root,"; }
+        if (flips & INDEX_PARENT_ROOT) { str += "index_parent_root,"; }
+        if (flips & INDEX_LEFT_SIBLING) { str += "index_left_sibling,"; }
+        if (flips & INDEX_RIGHT_SIBLING) { str += "index_right_sibling,"; }
+        return str;
+    }
+    void set_flip(uint32_t flip) { flips |= flip; }
+    void set_flip(std::string flip) {
+        if (flip == "index_parent_non_root") { set_flip(INDEX_PARENT_NON_ROOT); }
+        if (flip == "index_parent_root") { set_flip(INDEX_PARENT_ROOT); }
+        if (flip == "index_left_sibling") { set_flip(INDEX_LEFT_SIBLING); }
+        if (flip == "index_right_sibling") { set_flip(INDEX_RIGHT_SIBLING); }
+    }
+};
+
```

Comment:
I don't think these changes are needed any more since with new wb cache, we can directly set the flip.

Reply:
correct.